Automattic/wp-calypso
# Reader Lists: use data layer to get user's subscribed lists

#### Changes proposed in this Pull Request

Modernize the code used to fetch the user's subscribed lists, using data layer instead of thunks and `wpcom.undocumented()`.

#### Testing instructions



* Load Reader at http://calypso.localhost:3000/read. Ensure that your subscribed lists have been populated in the sidebar (if you have any):

<img width="249" alt="Screen Shot 2020-11-03 at 16 59 43" src="https://user-images.githubusercontent.com/17325/97948737-4dd69180-1df6-11eb-96df-2422c4d6de7a.png">

Ensure that a request to /read/lists was fired off during Reader load:

<img width="1684" alt="Screen Shot 2020-11-03 at 16 59 49" src="https://user-images.githubusercontent.com/17325/97948752-55963600-1df6-11eb-8eab-1e531b03852b.png">

No functional changes.
